src/libraries/System.Linq/src/System/Linq/ElementAt.cs

```diff
@@ -125,27 +102,44 @@ public static TSource ElementAt<TSource>(this IEnumerable<TSource> source, Index
             return element;
         }
 
-        private static bool TryGetElement<TSource>(IEnumerable<TSource> source, int index, [MaybeNullWhen(false)] out TSource element)
+        private static TSource? TryGetElementAt<TSource>(this IEnumerable<TSource> source, int index, out bool found) =>
```

Comment:
Curious what necessitated the shifting of the `TSource?` and `bool` return parameters?